hyperdrive.go

```diff
@@ -0,0 +1,194 @@
+package cloudflare
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/goccy/go-json"
+)
+
+var (
+	ErrMissingHyperdriveConfigID       = errors.New("required hyperdrive config id is missing")
+	ErrMissingHyperdriveConfigName     = errors.New("required hyperdrive config name is missing")
+	ErrMissingHyperdriveConfigPassword = errors.New("required hyperdrive config password is missing")
+)
+
+type HyperdriveConfig struct {
+	ID      string                  `json:"id,omitempty"`
+	Name    string                  `json:"name,omitempty"`
+	Origin  HyperdriveConfigOrigin  `json:"origin,omitempty"`
+	Caching HyperdriveConfigCaching `json:"caching,omitempty"`
+}
+
+type HyperdriveConfigOrigin struct {
+	Database string `json:"database,omitempty"`
+	Host     string `json:"host,omitempty"`
+	Port     int    `json:"port,omitempty"`
+	Scheme   string `json:"scheme,omitempty"`
+	User     string `json:"user,omitempty"`
+}
+
+type HyperdriveConfigCaching struct {
+	Disabled             bool `json:"disabled,omitempty"`
+	MaxAge               int  `json:"max_age,omitempty"`
+	StaleWhileRevalidate int  `json:"stale_while_revalidate,omitempty"`
+}
+
+type HyperdriveConfigListResponse struct {
+	Response
+	Result []HyperdriveConfig `json:"result"`
+}
+
+type CreateHyperdriveConfigParams struct {
+	Name     string                  `json:"name"`
+	Password string                  `json:"password"`
+	Origin   HyperdriveConfigOrigin  `json:"origin"`
+	Caching  HyperdriveConfigCaching `json:"caching,omitempty"`
+}
+
+type HyperdriveConfigResponse struct {
+	Response
+	Result HyperdriveConfig `json:"result"`
+}
+
+type UpdateHyperdriveConfigParams struct {
+	HyperdriveID string                  `json:"-"`
+	Name         string                  `json:"name"`
+	Password     string                  `json:"password"`
+	Origin       HyperdriveConfigOrigin  `json:"origin"`
+	Caching      HyperdriveConfigCaching `json:"caching,omitempty"`
+}
+
+// ListHyperdriveConfigs returns the Hyperdrive configs owned by an account.
+//
+// API reference: https://developers.cloudflare.com/api/operations/list-hyperdrive
+func (api *API) ListHyperdriveConfigs(ctx context.Context, rc *ResourceContainer) ([]HyperdriveConfig, error) {
```

Comment:
this method needs a third parameter per our [conventions](https://github.com/cloudflare/cloudflare-go/blob/master/docs/conventions.md)

Reply:
Added now.